src/app/frontend/replicasetlist/logsmenu_controller.js

```diff
@@ -80,4 +87,42 @@ export default class LogsMenuController {
         },
         (err) => { this.log_.error('Error fetching Replica Set pods: ', err); });
   }
+
+  /**
+   * @param {string} podName
+   * @param {string} containerName
+   * @return {string}
+   * @export
+   */
+  getLogsHref(podName, containerName) {
+    return this.state_.href(
+        logsStateName,
+        new StateParams(this.namespace, this.replicaSetName, podName, containerName));
+  }
+
+  /**
+   * Checks if pod contains at least one container. Return true if yes, otherwise false.
+   * @param {!backendApi.ReplicaSetPodWithContainers} pod
+   * @return {boolean}
+   * @export
+   */
+  podConteinerExists(pod) {
+    if (pod.podContainers[0].name === undefined) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Checks if pod containers were restarted. Return true if yes, otherwise false.
+   * @param {!backendApi.ReplicaSetPodWithContainers} pod
+   * @return {boolean}
+   * @export
+   */
+  podContainersRestarted(pod) {
```

Comment:
According to its type, pod cannot be null. I guess you should remove `!` from the type.